crates/ruff_diagnostics/src/diagnostic.rs

```diff
@@ -42,21 +42,22 @@ impl Diagnostic {
         }
     }
 
-    pub fn amend(&mut self, fix: Edit) -> &mut Self {
+    /// Set the [`Edit`] used to fix the diagnostic.
+    pub fn set_fix(&mut self, fix: Edit) {
```

Comment:
Alternatively, we could use `with_fix` and continue to return `self`, but we don't use that behavior anywhere anyway, so I made it a simple setter.